src/modules/webhook/webhookDispatcher.service.ts

```diff
@@ -0,0 +1,299 @@
+              messageContext: {
+                event: parsedMessage,
+                httpRequest: {
+                  url: webhook.url,
+                  startTime: startTime,
+                },
+                httpResponse: {
+                  data: responseData,
+                  statusCode: error.response.status,
+                },
+              },
+            });
+          } else if (error.request !== undefined) {
+            // Request was made but response was not received
+            this.logger.error({
+              message: 'Error sending event',
+              messageContext: {
+                event: parsedMessage,
+                httpRequest: {
+                  url: webhook.url,
+                  startTime: startTime,
+                },
+                httpResponse: null,
+                httpRequestError: {
+                  message: `Response not received. Error: ${error.message}`,
+                },
+              },
+            });
+          } else {
+            // Cannot make request
+            this.logger.error({
+              message: 'Error sending event',
+              messageContext: {
+                event: parsedMessage,
+                httpRequest: {
+                  url: webhook.url,
+                  startTime: startTime,
+                },
+                httpResponse: null,
+                httpRequestError: {
+                  message: error.message,
+                },
+              },
+            });
+          }
+          return of(undefined);
+        }),
+      ),
+    ).then((response: AxiosResponse | undefined) => {
+      if (response) {
+        webhook.incrementSuccess();
+        const endTime = Date.now();
+        const elapsedTime = endTime - startTime;
+        const responseData = this.parseResponseData(response.data);
+        this.logger.debug({
+          message: 'Success sending event',
+          messageContext: {
+            event: parsedMessage,
+            httpRequest: {
+              url: webhook.url,
+              startTime: startTime,
+              endTime: endTime,
+            },
+            httpResponse: {
+              data: responseData,
+              statusCode: response.status,
+              elapsedTimeMs: elapsedTime,
+            },
+          },
+        });
+      }
+      return response;
+    });
+  }
+
+  /**
+   * Evaluates the health of all webhooks by checking if any webhook has a consistently high failure rate.
+   * If a webhook exceeds the defined failure threshold within the allowed time window,
+   * it will be marked as disabled to prevent further issues.
+   */
+  async checkWebhooksHealth() {
+    this.logger.debug('Starting check webhooks health');
+    const activeWebhooks = this.getCachedActiveWebhooks();
+    const healthChecks = activeWebhooks.map(async (webhook) => {
+      if (
+        webhook.getTimeDelayedFromStartTime() >=
+        this.checkWebhookHealthWindowTime
```

Comment:
Window is only used to delay the check starts? Then it's not an sliding window, is more a wait for readiness. But it sounds good to me also for a first version

Reply:
Maybe we can rename the variable to make clear that's a delay? WDYF?